pkg/users/users.js

```diff
@@ -86,15 +87,18 @@ function AccountsPage() {
 
     const [details, setDetails] = useState(null);
     useInit(() => {
-        getLogins(shadow).then(setDetails);
+        getLogins().then(setDetails);
 
         // Watch `/var/run/utmp` to register when user logs in or out
-        const handle = cockpit.file("/var/run/utmp", { superuser: "try", binary: true });
-        handle.watch(() => {
-            getLogins(shadow).then(setDetails);
-        });
-        return handle;
-    }, [shadow], null, handle => handle.close());
+        const handleUtmp = cockpit.file("/var/run/utmp", { superuser: "try", binary: true });
+        handleUtmp.watch(() => getLogins().then(setDetails), { read: false });
+
+        // Watch /etc/shadow to register lock/unlock/expire changes; but avoid reading it, it's sensitive data
+        const handleShadow = cockpit.file("/etc/shadow", { superuser: "try" });
+        handleShadow.watch(() => getLogins().then(setDetails), { read: false });
+
+        return [handleUtmp, handleShadow];
+    }, [], null, handles => handles.forEach(handle => handle.close()));
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19751-20231215-131523-09b350cb-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)